### PR TITLE
Better fix for element-specific shortcut keys

### PR DESCRIPTION
**Pull request by wizofaus**
Resolves: https://github.com/musescore/MuseScore/issues/7687 etc.

For now, this reverts to MU3's mechanism of letting the current selected element have "first bite" at any keypress, and only if it doesn't process it, allow any actions that the shortcut key maps to to fire.

- [ x ] I signed [CLA](https://musescore.org/en/cla)
- [ x ] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [ x ] I made sure the code compiles on my machine
- [ x ] I made sure there are no unnecessary changes in the code
- [ x ] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [ x ] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [ x ] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [  ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/notation/view/notationpaintview.cpp [hunk 1]
```diff
@@ -652,6 +652,7 @@ void NotationPaintView::mousePressEvent(QMouseEvent* event)
 {
     setFocus(true);
     emit activeFocusRequested();
+    forceActiveFocus();
```
Comment (wizofaus) [inline src/notation/view/notationpaintview.cpp:655]: Necessary or NotationPaintView won't receive ShortcutOverride events at all. Arguably it should forceActiveFocus the moment it's first displayed too?

File: src/notation/view/notationpaintview.cpp [hunk 2]
```diff
@@ -688,30 +689,19 @@ void NotationPaintView::hoverMoveEvent(QHoverEvent* event)
...
-            }
-            if (event->modifiers() & Qt::KeyboardModifier::ControlModifier) {
-                sequence = "Ctrl+" + sequence;
-            }
-            const auto& shortcuts = shortcutsRegister()->shortcutsForSequence(sequence);
-            const std::shared_ptr<IUiActionsRegister> theRegister = actionsRegister();
-            if (std::find_if(shortcuts.begin(), shortcuts.end(), [theRegister](auto s) {
-                return theRegister->actionState(s.action).enabled;
-            }) != shortcuts.end()) {
-                return;
```
Comment (wizofaus) [inline src/notation/view/notationpaintview.cpp:710]: there's still an argument for keeping this logic if we decide it makes more sense to have the shortcut-mapped actions to kick in first, but that requires a lot more work to ensure they're only enabled when they make sense (e.g. "next-chord" shouldn't be enabled when a slur is selected etc. etc.)

File: src/notation/view/notationviewinputcontroller.cpp [hunk 3]
```diff
@@ -473,11 +473,9 @@ void NotationViewInputController::hoverMoveEvent(QHoverEvent* event)
     }
 }
 
-void NotationViewInputController::keyReleaseEvent(QKeyEvent* event)
+void NotationViewInputController::keyPressEvent(QKeyEvent* event)
 {
-    if (m_view->notationInteraction()->isTextEditingStarted()) {
-        m_view->notationInteraction()->editText(event);
-    }
+    m_view->notationInteraction()->editText(event);
```
Comment (wizofaus) [inline src/notation/view/notationviewinputcontroller.cpp:478]: 'editText' is a bit of a misnomer now, probably should rename to "keyPressEvent".

File: src/notation/internal/notationinteraction.cpp [hunk 4]
```diff
@@ -1937,18 +1937,26 @@ void NotationInteraction::startEditText(Element* element, const PointF& cursorPo
 
 void NotationInteraction::editText(QKeyEvent* event)
 {
-    IF_ASSERT_FAILED(m_textEditData.element) {
+    if (!m_textEditData.element && selection()->element()) {
```
Comment (wizofaus) [inline src/notation/internal/notationinteraction.cpp:1940]: should probably rename to just m_editData now we're using it for other types of elements too

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**